### PR TITLE
[OF-1335] feat: Added HotspotSpaceComponent

### PR DESCRIPTION
**Pull request by MAG-AW**
* Added the new Component
* Added script bindings
* Added tests

*Removed duplicated TextSpaceComponent script bindings

### CODE REVIEW

File: Library/include/CSP/Multiplayer/Components/HotspotSpaceComponent.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+	/// @brief Gets the Name of the Hotspot.
+	/// @return The Name of the Hotspot.
+	const csp::common::String& GetName() const;
+
+	/// @brief Sets the Name of this Hotspot.
+	/// @param Value The Name of this Hotspot.
+	void SetName(const csp::common::String& Value);
+
+    HotspotType GetHotspotType() const;
```
Comment (MAG-AdamThorn) [inline Library/include/CSP/Multiplayer/Components/HotspotSpaceComponent.h:70]: Missing documentation for for get/set hotspot type.

File: Tests/src/PublicAPITests/ComponentTests/HotspotComponentTests.cpp [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+		[](csp::multiplayer::SpaceEntity* Entity)
+		{
+		});
+
+	// Create object to represent the text
+	csp::common::String ObjectName = "Object 1";
+	SpaceTransform ObjectTransform = {csp::common::Vector3::Zero(), csp::common::Vector4::Zero(), csp::common::Vector3::One()};
+	auto [CreatedObject]		   = AWAIT(EntitySystem, CreateObject, ObjectName, ObjectTransform);
+
+	// Create text component
```
Comment (MAG-AdamThorn) [inline Tests/src/PublicAPITests/ComponentTests/HotspotComponentTests.cpp:88]: The comment should say 'hotspot'

**Review by MAG-AdamThorn — CHANGES_REQUESTED** [review]

File: Library/include/CSP/Multiplayer/Components/HotspotSpaceComponent.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+#include "CSP/Multiplayer/Components/Interfaces/IRotationComponent.h"
+#include "CSP/Multiplayer/Components/Interfaces/IVisibleComponent.h"
+
+
+namespace csp::multiplayer
+{
+
+enum class HotspotType
+{
+	DEFAULT = 0,
```
[the remark this answers is not in the document]
Comment (MAG-AdamThorn) [reply Library/include/CSP/Multiplayer/Components/HotspotSpaceComponent.h:35]: That was likely missed when we moved away from the idea of using the POI System. I think you can remove it unless you are aware of any reason to keep it.

**Review by MAG-AdamThorn — APPROVED** [review]

File: Tests/src/PublicAPITests/ComponentTests/HotspotComponentTests.cpp [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+	EXPECT_EQ(HotspotComponent->GetPosition().Y, 0.0f);
+	EXPECT_EQ(HotspotComponent->GetPosition().Z, 0.0f);
+	EXPECT_EQ(HotspotComponent->GetComponentType(), ComponentType::Hotspot);
+	EXPECT_EQ(HotspotComponent->GetIsARVisible(), true);
+	EXPECT_EQ(HotspotComponent->GetIsVisible(), true);
+	EXPECT_EQ(HotspotComponent->GetRotation().W, 1);
+	EXPECT_EQ(HotspotComponent->GetRotation().X, 0);
+	EXPECT_EQ(HotspotComponent->GetRotation().Y, 0);
+	EXPECT_EQ(HotspotComponent->GetRotation().Z, 0);
+	EXPECT_EQ(HotspotComponent->GetHotspotType(), HotspotType::DEFAULT);
```
[the remark this answers is not in the document]
Comment (MAG-AdamThorn) [reply Tests/src/PublicAPITests/ComponentTests/HotspotComponentTests.cpp]: Yes... yes I am!
Thanks

File: Library/src/Multiplayer/Components/HotspotSpaceComponent.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+{
+	return static_cast<HotspotType>(GetIntegerProperty(static_cast<uint32_t>(HotspotPropertyKeys::HotspotType)));
+}
+
+void HotspotSpaceComponent::SetHotspotType(HotspotType Value)
+{
+	SetProperty(static_cast<uint32_t>(HotspotPropertyKeys::HotspotType), static_cast<int64_t>(Value));
+}
+
+const csp::common::String& HotspotSpaceComponent::GetUniqueComponentId()
```
Comment (MAG-mv) [inline Library/src/Multiplayer/Components/HotspotSpaceComponent.cpp:58]: Why is UniqueComponentId a property of the class? Is it worth removing this and making the function const?

**Review by MAG-mv — APPROVED** [review]

**Review by MAG-AdamThorn — APPROVED** [review]

**Pull request merged**